tests/TypeSystemTest/BaseTest.cs

```diff
@@ -0,0 +1,197 @@
+    /// </summary>
+    /// <param name="fullName"></param>
+    /// <returns></returns>
+    private static string GetNormalizzeFullName(string fullName)
+    {
+        if (!_cleanedFullName.TryGetValue(fullName, out var cleanedName))
+        {
+            cleanedName = fullName;
+            var index = cleanedName.IndexOf('`');
+            if (index > -1)
+            {
+                var fullNameLen = fullName.Length;
+                var sb = new StringBuilder(fullNameLen);
+                var numberOfArgs = 0;
+                int i = 1;
+                for (; index + i < fullNameLen && char.IsNumber(fullName[index + i]); i++)
+                {
+                    numberOfArgs = (numberOfArgs * (int)Math.Pow(i, 10)) + (fullName[index + i] - 48);
+                }
+                sb.Append(fullName, 0, index + i);
+                var typeArgStart = index + i + 1;
+                if (numberOfArgs > 0 && typeArgStart < fullNameLen)
+                {
+                    sb.Append('<');
+                    var argIndex = 0;
+                    var skip = false;
+                    var parsingArg = false;
+                    for (i = typeArgStart; argIndex < numberOfArgs && i < fullNameLen; i++)
+                    {
+                        var c = fullName[i];
+                        switch (c)
+                        {
+                            case '[' when !parsingArg:
+                                parsingArg = true;
+                                skip = false;
+                                break;
+                            case ',' when parsingArg && !skip:
+                                argIndex++;
+                                sb.Append(',');
+                                skip = true;
+                                break;
+                            case ',' when !parsingArg:
+                                skip = true;
+                                break;
+                            case ']' when skip:
+                                skip = false;
+                                parsingArg = false;
+                                break;
+                            default:
+                                if (!skip)
+                                {
+                                    sb.Append(c);
+                                }
+                                break;
+                        }
+                    }
+                    sb[sb.Length - 1] = '>';
+                }
+                cleanedName = sb.ToString();
+            }
+            _cleanedFullName[fullName] = cleanedName;
+        }
+        return cleanedName;
+    }
+
+    protected BaseTest(IXamlTypeSystem typeSystem)
+    {
+        _typeSystem = typeSystem;
+    }
+
+    protected IXamlTypeSystem TypeSystem => _typeSystem;
+
+    [Fact]
+    public void Generic_Types_Are_Correctly_Handled()
+    {
+        var wut = TypeSystem.GetType("TypeSystemTest.Models.TestType");
```

Comment:
As much as I like short variable names, they make the code way less readable when coming in such abundance. Can we have a bit more descriptive ones?